src/parser/jobs/ast/modules/Sect/Sect.tsx

```diff
@@ -1,163 +1,90 @@
-		let aspectedCast: CastEvent | null = null
-
-		for (const event of events) {
-			if (event.type === 'applybuff' && SECT_STATUSES.includes(event.ability.guid)) {
-				// They started the fight without a sect and switched it on mid-fight :blobsweat: we gud here
-				break
-			}
+	//this section sets up the statuses using this.data.status
+	private sectStatuses: Array<Status['id']> = []
+	private diurnalSectStatuses: Array<Status['id']> = []
+	private nocturnalSectStatuses: Array<Status['id']> = []
 
-			if (!aspectedCast && event.type === 'cast' && ASPECTED_ACTIONS.includes(event.ability.guid)) {
-				// Detecting a cast of an aspected action, so now we examine what comes out of it
-				aspectedCast = event
-
-			} else if (aspectedCast
-				&& (event.type === 'applybuff' || event.type === 'refreshbuff') && [...NOCTURNAL_SECT_STATUSES, ...DIURNAL_SECT_STATUSES].includes(event.ability.guid)) {
-				// This is an applybuff event of a sect buff that came after an aspected action
-
-				if (this.mapCastToBuff(aspectedCast.ability.guid).includes(event.ability.guid)
-					&& [...DIURNAL_SECT_STATUSES, ...NOCTURNAL_SECT_STATUSES].includes(event.ability.guid)) {
-
-					const SECT_ABILITY = DIURNAL_SECT_STATUSES.includes(event.ability.guid) ? DIURNAL_SECT_BUFF_ABILITY : NOCTURNAL_SECT_BUFF_ABILITY
-
-					// Fab a sect buff event at the start of the fight
-					events.splice(0, 0, {
-						...event,
-						timestamp: startTime - 1,
-						type: 'applybuff',
-						ability: SECT_ABILITY,
-						targetID: event.sourceID,
-						targetIsFriendly: true,
-					})
-					break
-				}
-			} else {
-				continue
-			}
+	override initialise() {
 
-		}
+		this.sectStatuses = SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
+		this.diurnalSectStatuses = DIURNAL_SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
+		this.nocturnalSectStatuses = NOCTURNAL_SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
 
-		return events
+		//this section uses hooks to go through the various functions
+		this.addEventHook(filter<Event>()
+			.source(this.parser.actor.id)
+			.type('statusApply')
+			.status(oneOf([...this.sectStatuses, ...this.nocturnalSectStatuses, ...this.diurnalSectStatuses]))
+		, this.onApplyStatus)
+		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onCast(event: CastEvent) {
-		// If they used a sect after the fight started, it means they pulled without one on; otherwise, we probably added it in normalize.
-		this.pullWithoutSect = event.timestamp >= this.parser.fight.start_time
+	//to look for the sect buff or any relating sects statuses we would expect
+	private onApplyStatus(event: Events['statusApply']) {
 
-		const sect = getDataBy(ACTIONS, 'id', event.ability.guid)
-		if (sect) {
-			this.activeSectId = sect.id
+		//if they switched it mid-fight, this section will pick it up
+		if (this.sectStatuses.includes(event.status)) {
+			if (this.activeSectId !== 0) {
```

Comment:
Prefer initialising as undefined and `!= null` here, it's clearer what the intent is.

Reply:
updated to != null or == null where appropriate throughout based on this comment